### PR TITLE
refactor: apply webpack-defaults

### LINKED ISSUE #47: webpack-defaults upgrade
Addition of `webpack-defaults` & associated refactoring as a part of the next `Major` release

Issue exists for status tracking across the organization. 

>Please do not close

### PR DESCRIPTION
**Pull request by d3viant0ne**
Intended to be merged & released as a part of `1.0.0` on a beta `dist-tag` once this has been finished and properly tested.

BREAKING CHANGE: 

Enforces NodeJS > 4.3 via engines 

Closes #47

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+import loaderUtils from 'loader-utils';
+
+export default function (source) {
+  let value = typeof source === 'string' ? JSON.parse(source) : source;
+  const options = loaderUtils.getOptions(this) || {};
+  value = JSON.stringify(value)
+    .replace(/\u2028/g, '\\u2028')
+    .replace(/\u2029/g, '\\u2029');
+  value = options.stringify ? `'${value}'` : value;
+  const module = this.version && this.version >= 2 ? `export default ${value};` : `module.exports = ${value};`;
```
Comment (michael-ciniawsky) [inline src/index.js]: webpack 2 has JSON support in core, so better display a warning to about that && remove `json-loader` from `webpack.config.js`

 ```js
if (this.version && this.version < 2)
   const module = `module.exports = ${value};`;
   return cb(null, module)
}

this.emitWarning(`message`)
return cb(null, value) // not 100% about that :)
```

**Comment by d3viant0ne** [comment]
So given webpack > 2 has native json support, we should probably skip the defaults upgrade and just throw an deprecation warning for 2.x.

The defaults upgrade is of no benefit to webpack 1.x.

**Comment by michael-ciniawsky** [comment]
Yep or that I can do a separate PR with the warning etc

**Comment by michael-ciniawsky** [comment]
I think we should also check `json5-loader` in terms of webpack 2 compatibility, here @json-loader I closed a few issue because of `json-loader` usage + webpack 2 core JSON support

**Pull request closed without merging**